devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy

```diff
@@ -0,0 +1,130 @@
+    def resultClosed = '0x2000'
+    def resultBatteryNew = '0x1000'
+    def resultBatteryOut = '0x0800'
+    if (zs.isAlarm1Set()) {
+        runIn(seconds02, stopMotion) 
+        resultMap = getMotionResult('active')  
+    } else if (zs.isBatterySet()){
+        resultMap = getBatteryResult(10)
+    } else if (description.contains(resultOccupied)) {
+        runIn(seconds01, resetOccupancy)    
+        resultMap = getMotionResult('occupied') 
+    } else if (description.contains(resultOpenned)) {
+        resultMap = getMotionResult('openned') 
+    } else if (description.contains(resultClosed)) {    
+        resultMap = getMotionResult('closed')           
+    } else if (description.contains(resultBatteryNew)) {
+        resultMap = getBatteryResult(100)
+    } else if (description.contains(resultBatteryOut)) {
+        resultMap = getBatteryResult(0)
+    }  
+}
+
+private Map getBatteryResult(rawValue) {
+    log.debug "Battery rawValue = ${rawValue}"
+    createEvent(name:"battery", value:rawValue)
+}
+
+private Map getMotionResult(value) {
+    if (value == 'active') {
+        log.debug 'intrusion'
+        String descriptionText = "{{ device.displayName }} detected intrusion"
+        createEvent(
+            name: 'motion',
+            value: value,
+            descriptionText: descriptionText,
+            translatable: false
+        )
+    } else if (value == 'occupied') {
+        log.debug 'detected occupancy'
+        String descriptionText = "{{ device.displayName }} detected occupant"
+        createEvent(
+            name: 'presence',
+            value: "present",
+            descriptionText: descriptionText,
+            translatable: false
+        )
+    } else if (value == 'openned') {
+        log.debug 'detected window openned'
+        String descriptionText = "{{ device.displayName }} detected window openned"
+        createEvent(
+            name: 'contact',
+            value: "open",
+            descriptionText: descriptionText,
+            translatable: false
+        )
+    } else if (value == 'closed') {
+        log.debug 'detected  window closed'
+        String descriptionText = "{{ device.displayName }} detected window closed"
+        createEvent(
+            name: 'contact',
+            value: "closed",
+            descriptionText: descriptionText,
+            translatable: false
+        )
+    }
+}
+
+def installed() {
+    initialize()
+}
+
+def initialize() {
+    sendEvent(name:"motion", value:"inactive")
+    sendEvent(name:"battery", value:"100")
+    sendEvent(name:"presence", value:"not present")
+    sendEvent(name:"contact", value:"closed")
+}
+
+def stopMotion() {
+    def seconds = 120
+    if (device.currentState('motion')?.value == "active") {
+        sendEvent(name:"motion", value:"inactive", isStateChange: true)
+        log.debug "${device.displayName} reset to monitoring after ${seconds} seconds"
+    }
+}
+
+def resetOccupancy() {
+    def seconds = 60
```

Comment:
If You use this variable in log message only - wouldn't it be better to write that value explicitly in that message ?